### PR TITLE
spotbugs rule for direct instantiation of BinaryDecoder

### PR DESCRIPTION
**Pull request by radai-rosenblatt**

### CODE REVIEW

File: spotbugs-plugin/src/test/java/com/linkedin/avroutil1/spotbugs/GoodClass.java [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
 import java.io.OutputStream;
 
 public class GoodClass {
 
     public void instantiateBinaryEncoder() {
         BinaryEncoder bobTheEncoder = AvroCompatibilityHelper.newBinaryEncoder( (OutputStream) null);
     }
+
+    public void instantiateBinaryDecoder() {
+        BinaryDecoder bobTheDecoder = AvroCompatibilityHelper.newBinaryDecoder( (InputStream) null);
```
Comment (FelixGV) [inline spotbugs-plugin/src/test/java/com/linkedin/avroutil1/spotbugs/GoodClass.java:18]: So... what does spotbugs say if you pass null without casting it? Does it think it's bad?

File: spotbugs-plugin/src/test/java/com/linkedin/avroutil1/spotbugs/GoodClass.java [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
+        BinaryDecoder robertTheDecoder = DecoderFactory.defaultFactory().createBinaryDecoder(new byte[] {1, 2, 3}, null);
```
Comment (FelixGV) [inline spotbugs-plugin/src/test/java/com/linkedin/avroutil1/spotbugs/GoodClass.java:19]: Why is this considered good? Won't it fail in some Avro versions and therefore be non-portable?

Comment (radai-rosenblatt) [reply spotbugs-plugin/src/test/java/com/linkedin/avroutil1/spotbugs/GoodClass.java:19]: this particular invocation would be ambiguous and wont compile without the cast

File: spotbugs-plugin/src/test/java/com/linkedin/avroutil1/spotbugs/GoodClass.java (hunk 1 above)
Comment (radai-rosenblatt) [reply spotbugs-plugin/src/test/java/com/linkedin/avroutil1/spotbugs/GoodClass.java:18]: this particular invocation would be ambiguous and wont compile without the cast

File: spotbugs-plugin/src/test/java/com/linkedin/avroutil1/spotbugs/GoodClass.java (hunk 2 above)
Comment (radai-rosenblatt) [reply spotbugs-plugin/src/test/java/com/linkedin/avroutil1/spotbugs/GoodClass.java:19]: this particular invocation actually works across avro 1.4-10.
its deprecated in ~1.6+, but is actually portable.
also i wanted to test for something that isnt a helper invocation

Comment (FelixGV) [reply spotbugs-plugin/src/test/java/com/linkedin/avroutil1/spotbugs/GoodClass.java:19]: Ok, interesting...

**Review by FelixGV — APPROVED** [review]

**Review by navina — APPROVED** [review]
lgtm!

**Pull request merged**